network/src/protocols/discovery/test.rs

```diff
@@ -317,9 +338,91 @@ fn outbound() {
+    let seed_peer_payload = gen_full_node_payload();
+    let seed_peer_id = PeerId::random();
+    let (seed_pub_keys, seed_signer) = generate_network_pub_keys_and_signer(seed_peer_id);
+    let trusted_peers = Arc::new(RwLock::new(
+        vec![(seed_peer_id, seed_pub_keys), (peer_id, self_pub_keys)]
+            .into_iter()
+            .collect(),
+    ));
+    // Setup discovery.
+    let (_, mut conn_mgr_reqs_rx, mut peer_mgr_notifs_tx, _) = setup_discovery(
+        &mut rt,
+        peer_id,
+        addrs,
+        seed_peer_id,
+        seed_peer_info,
+        self_signer,
+        trusted_peers,
+    );
+
+    // Fake connectivity manager and dialer.
+    let f_peer_mgr = async move {
+        // Connectivity manager receives addresses of the seed peer during bootstrap.
+        expect_address_update(&mut conn_mgr_reqs_rx, seed_peer_id, &seed_peer_addrs[..]).await;
+
+        // Notify discovery actor of inbound substream.
+        let (dialer_substream, listener_substream) = MemorySocket::new_pair();
+        peer_mgr_notifs_tx
+            .send(PeerManagerNotification::NewInboundSubstream(
+                seed_peer_id,
+                NegotiatedSubstream {
+                    protocol: ProtocolId::from_static(DISCOVERY_PROTOCOL_NAME),
+                    substream: listener_substream,
+                },
+            ))
+            .await
+            .unwrap();
+        // Wrap dialer substream in a framed substream.
+        let mut dialer_substream =
+            Framed::new(dialer_substream.compat(), UviBytes::<Bytes>::default()).sink_compat();
+
+        // Send DiscoveryMsg consisting of the new seed peer's discovery note.
+        // The discovery actor should send the addrs in the new seed peer note
+        // _and_ the configured seed addrs to the connectivity manager.
+        let new_seed_addrs = vec![Multiaddr::from_str("/ip4/127.0.0.1/tcp/9091").unwrap()];
+        let new_seed_info = create_peer_info(new_seed_addrs.clone());
+        let seed_note = create_note(&seed_signer, seed_peer_id, new_seed_info, seed_peer_payload);
+        let mut msg = DiscoveryMsg::new();
+        msg.mut_notes().push(seed_note.clone());
+        dialer_substream
+            .send(msg.write_to_bytes().unwrap().into())
+            .await
+            .unwrap();
+
+        // The addrs sent to connectivity manager should also include the
+        // configured seed peer addrs for seed-peer-received notes.
+        let mut expected_seed_addrs = new_seed_addrs.clone();
+        expected_seed_addrs.append(&mut seed_peer_addrs);
+
+        // Connectivity manager receives an update including the seed peer's new
+        // note addrs and the original configured seed addrs.
+        expect_address_update(
+            &mut conn_mgr_reqs_rx,
+            seed_peer_id,
+            &expected_seed_addrs[..],
+        )
+        .await;
+    };
+    rt.block_on(f_peer_mgr.boxed().unit_error().compat())
+        .unwrap();
```

Comment:
if you didn't modify the old code, would this fail here?

(this code is foreign to me, would like to understand how this test is actually being evaluated)

Reply:
Yep! Without the changes in `discovery/mod.rs`, this line:

```rust
// L419:
expect_address_update(
    &mut conn_mgr_reqs_rx,
    seed_peer_id,
    &expected_seed_addrs[..],
)
.await;
```

would fail, since the seed peer addresses would not be included in the `UpdateAddresses` message sent to the (mocked) connectivity manager.